File: tools/reset_store_logo_db.py
import os
from pymongo import MongoClient

def main():
    env_path = os.path.join(os.path.dirname(__file__), '..', 'backend', '.env')
    
    # Read .env file manually
    env_vars = {}
    with open(env_path, 'r', encoding='utf-8') as f:
        for line in f:
            if '=' in line:
                key, val = line.split('=', 1)
                env_vars[key.strip()] = val.strip()

    mongodb_url = env_vars.get('MONGODB_URL', 'mongodb://localhost:27017')
    database_name = env_vars.get('DATABASE_NAME', 'Mian_Sons_hardware_db')

    print(f"Connecting to database: {database_name}...")
    client = MongoClient(mongodb_url)
    try:
        db = client[database_name]
        settings_col = db['settings']
        
        # Check if store_info exists
        doc = settings_col.find_one({'_id': 'store_info'})
        if doc:
            print("Found store_info document. Current value:")
            print(doc)
            
            # Update logoUrl inside the document
            # If the schema stores it directly or under a 'value' dict, we update both to be safe
            update_fields = {'logoUrl': '/images/store-logo.png'}
            if 'value' in doc and isinstance(doc['value'], dict):
                update_fields['value.logoUrl'] = '/images/store-logo.png'
                
            res = settings_col.update_one({'_id': 'store_info'}, {'$set': update_fields})
            print(f"Update result: matched={res.matched_count}, modified={res.modified_count}")
            print("Successfully updated database logo settings to use default '/images/store-logo.png'.")
        else:
            print("store_info document not found in settings collection.")
            
    except Exception as e:
        print("Failed to update database:", e)
    finally:
        client.close()

if __name__ == '__main__':
    main()

File: tools/reset_store_logo_db.js
const { MongoClient } = require('mongodb');
const fs = require('fs');
const path = require('path');

// Read env file from backend
const envPath = path.join(__dirname, '..', 'backend', '.env');
const envContent = fs.readFileSync(envPath, 'utf8');
const envVars = {};
envContent.split('\n').forEach(line => {
  const parts = line.split('=');
  if (parts.length >= 2) {
    const key = parts[0].trim();
    const value = parts.slice(1).join('=').trim();
    envVars[key] = value;
  }
});

const url = envVars.MONGODB_URL || 'mongodb://localhost:27017';
const dbName = envVars.DATABASE_NAME || 'Mian_Sons_hardware_db';

async function main() {
  const client = new MongoClient(url);
  try {
    await client.connect();
    console.log('Connected to MongoDB database...');
    const db = client.db(dbName);
    const settingsCollection = db.collection('settings');
    
    // Find store_info setting
    const storeInfoDoc = await settingsCollection.findOne({ _id: 'store_info' });
    if (storeInfoDoc) {
      console.log('Found store_info document. Current logoUrl length:', storeInfoDoc.logoUrl ? storeInfoDoc.logoUrl.length : 'undefined');
      
      // Update logoUrl to /images/store-logo.png
      await settingsCollection.updateOne(
        { _id: 'store_info' },
        { 
          $set: { 
            logoUrl: '/images/store-logo.png',
            'value.logoUrl': '/images/store-logo.png' // just in case it's nested under value
          } 
        }
      );
      console.log('Successfully updated logoUrl to "/images/store-logo.png" in database!');
    } else {
      console.log('store_info document not found in settings collection.');
    }
  } catch (err) {
    console.error('Database operation failed:', err);
  } finally {
    await client.close();
  }
}

main();
